src/components/AddForm.js

```diff
@@ -1,49 +1,56 @@
 import React, { useState } from 'react';
+import { connect } from 'react-redux';
+
+import { addSmurf } from '../actions';
+
+const initialForm = {
+    name: "",
+    position: "",
+    nickname: "",
+    description: ""
+}
 
 const AddForm = (props) => {
-    const [state, setState] = useState({
-        name:"",
-        position:"",
-        nickname:"",
-        description:""
-    });
+    const { errorMessage, addSmurf } = props
 
-    //remove when error state is added
-    const errorMessage = "";
+    const [state, setState] = useState(initialForm);
 
     const handleChange = e => {
         setState({
             ...state,
-            [e.target.name]:e.target.value
+            [e.target.name]: e.target.value
         });
     }
 
     const handleSubmit = e => {
         e.preventDefault();
         if (state.name === "" || state.position === "" || state.nickname === "") {
-            //dispatch a custom error action
-        } else {
-            //dispatch an addSmurf action
+            addSmurf(state)
```

Comment:
In this case at least one form field is empty and it looks like they want you to dispatch an error message here. The addSmurf could potentially return an error but that requires talking to the BE which takes time when you could return the error immediately here.